classes/Hook/HookDisplayOrderConfirmation.php

```diff
@@ -0,0 +1,114 @@
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\Module\Ps_Googleanalytics\Hooks;
+
+use Cart;
+use Configuration;
+use Context;
+use PrestaShop\Module\Ps_Googleanalytics\GoogleAnalyticsTools;
+use PrestaShop\Module\Ps_Googleanalytics\Handler\GanalyticsJsHandler;
+use PrestaShop\Module\Ps_Googleanalytics\Repository\GanalyticsRepository;
+use PrestaShop\Module\Ps_Googleanalytics\Wrapper\ProductWrapper;
+use Ps_Googleanalytics;
+use Shop;
+use Validate;
+
+class HookDisplayOrderConfirmation implements HookInterface
+{
+    private $module;
+    private $context;
+    private $params;
+
+    public function __construct(Ps_Googleanalytics $module, Context $context)
+    {
+        $this->module = $module;
+        $this->context = $context;
+    }
+
+    /**
+     * run
+     *
+     * @return string
+     */
+    public function run()
+    {
+        if (true === $this->module->psVersionIs17) {
+            $order = $this->params['order'];
+        } else {
+            $order = $this->params['objOrder'];
+        }
+
+        if (Validate::isLoadedObject($order) && $order->getCurrentState() != (int) Configuration::get('PS_OS_ERROR')) {
+            $ganalyticsRepository = new GanalyticsRepository();
+            $gaOrderSent = $ganalyticsRepository->findGaOrderByOrderId((int) $order->id);
+
+            if (false === $gaOrderSent) {
+                $ganalyticsRepository->addNewRow(
+                    [
+                        'id_order' => (int) $order->id,
+                        'id_shop' => (int) $this->context->shop->id,
+                        'sent' => 0,
+                        'date_add' => 'NOW()',
+                    ]
+                );
+
+                if ($order->id_customer == $this->context->cookie->id_customer) {
+                    $orderProducts = [];
+                    $cart = new Cart($order->id_cart);
+                    $gaTools = new GoogleAnalyticsTools();
+                    $gaTagHandler = new GanalyticsJsHandler($this->module, $this->context);
+                    $productWrapper = new ProductWrapper($this->context);
+
+                    foreach ($cart->getProducts() as $order_product) {
+                        $orderProducts[] = $productWrapper->wrapProduct($order_product, [], 0, true);
+                    }
+
+                    $gaScripts = 'MBG.addCheckoutOption(3,\'' . $order->payment . '\');';
+                    $transaction = [
+                        'id' => $order->id,
+                        'affiliation' => (Shop::isFeatureActive()) ? $this->context->shop->name : Configuration::get('PS_SHOP_NAME'),
+                        'revenue' => $order->total_paid,
```

Comment:
What about currencies ? If a shop has 2 currencies enabled, two orders with `revenue:200` are not the same, 200 USD is not the same as 200 EUR

Reply:
Agree with that too. The code was like that before.
I'm not able to confirm that it's an error or not :\

Reply:
YOLO then 😄